cupyx/scipy/sparse/data.py

```diff
@@ -139,15 +140,28 @@ def _min_or_max_axis(self, axis, min_or_max, sum_duplicates, nonzero):
         N = self.shape[axis]
         if N == 0:
             raise ValueError("zero-size array to reduction operation")
+        M = self.shape[1 - axis]
 
         mat = self.tocsc() if axis == 0 else self.tocsr()
         if sum_duplicates:
             mat.sum_duplicates()
 
         # Do the reudction
         value = mat._minor_reduce(min_or_max, axis, nonzero)
+        major_index = cupy.arange(M)
 
-        return value
+        mask = value != 0
+        major_index = cupy.compress(mask, major_index)
```

Comment:
OK to leave the following issues because the PR is a bugfix.

- For efficiency, it is possible that `mat._minor_reduce` should compute the `major_index` and the compressed `value`.
- `coo_matrix(D)` could be implemented, which is discussed at https://github.com/cupy/cupy/issues/3289#issuecomment-653194583

Reply:
I'll revisit these points in another PR!